src/gui/DatabaseTabWidget.cpp

```diff
@@ -128,20 +128,18 @@ void DatabaseTabWidget::openDatabase(const QString& fileName, const QString& pw,
     DatabaseManagerStruct dbStruct;
 
     // test if we can read/write or read the file
-    QFile file(fileName);
     // TODO: error handling
-    if (!file.open(QIODevice::ReadWrite)) {
-        if (!file.open(QIODevice::ReadOnly)) {
+    if (!fileInfo.isReadable() && !fileInfo.isWritable()) {
+        if (fileInfo.isReadable()) {
```

Comment:
This doesn't make sense:
The file can't be readable in this branch as you checked !isReadable() before.

isReadable()/isWriteable() is not good way to check this anyway.
You can only reliable check if a file is readable/writeable if you open it that way.
For example I doubt those methods check for read-only filesystem.
Additionally MACs like SELinux and AppArmor can deny access to files without you being able to know beforehand.